File: web-app/src/types/flight.ts
export interface Flight {
    id: number | null;
    number: string;
    fromCity: string;
    toCity: string;
    departure: string; // ISO string
    arrival: string; // ISO string
    passengerCount: number;
    ticketPrice: number;
    createdAt?: string;
    updatedAt?: string;
}

export interface FlightCreate {
    number: string;
    fromCity: string;
    toCity: string;
    departure: string;
    arrival: string;
    passengerCount: number;
    ticketPrice: number;
}

export interface FlightUpdate {
    id: number;
    number: string;
    fromCity: string;
    toCity: string;
    departure: string;
    arrival: string;
    ticketPrice: number;
}

export interface FlightQuery {
    number?: string;
    fromCity?: string;
    toCity?: string;
    departureFrom?: string;
    departureTo?: string;
}

export interface FlightDelete {
    id: number;
}

File: FlightManagement/src/main/java/io/github/bagdad/flightmanagement/model/Flight.java
package io.github.bagdad.flightmanagement.model;


import lombok.Builder;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.time.OffsetDateTime;

@Table("flights")
@RequiredArgsConstructor
@Getter
@Setter
public class Flight {

    @Id
    private Long id;

    private String number;

    private String fromCity;

    private String toCity;

    private OffsetDateTime departure;

    private OffsetDateTime arrival;

    private Integer passengerCount;

    private BigDecimal ticketPrice;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    public boolean canBook(Integer passengerCount) {
        return this.passengerCount >= passengerCount;
    }

    public void book(Integer passengerCount) {
        this.passengerCount -= passengerCount;
    }

    public void cancelReservation(Integer passengerCount) {
        this.passengerCount += passengerCount;
    }

}
